events-api/src/modules/events/events.controller.ts

```diff
@@ -23,3 +29,64 @@ EventsController.get(
     });
   }
 );
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsCreateDtoSchema),
+  async (req, res) => {
+    await EventsService.createEvent(req.body);
+
+    return res.status(201).json({
+      message: "Event created successfully",
+    });
+  }
+);
+
+EventsController.patch(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  validateRequestBody(eventsUpdateDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.updateEvent(
+      req.body,
+      req.params["id"] as string
+    );
+    if (event === null) {
```

Comment:
What does event null means? Can it also mean that update failed?

Reply:
it was just not avaliable and I also have question can I spesific errors from the repository? 
then handle it in service?

Reply:
![image](https://github.com/user-attachments/assets/bd451e7c-5847-46a6-8045-5ee445d8ebee)

Reply:
It depens on the approach and should be consistent within the project. I would suggest that:

- Check if the event is in the DB. If not throw an error (400) 
- Update the record and return the updated record

Having just null gives a lot of ambiguity for interpretation

Reply:
![image](https://github.com/user-attachments/assets/96798822-9ca9-4543-8002-99dfec0fd95f)
handling not found error in repo then catching it in controller
![image](https://github.com/user-attachments/assets/1e56109b-06f9-43f7-a43e-2e8e986a3d8e)

Reply:
I think that error should not be thrown in the repository. Service is a better place for this

Reply:
I think that error should not be thrown in the repository. Service is a better place for this.